File: Assignment_26/assignment5.c
/////////////////////////////////////////////////////////////////////////////////
//
//  File name :     assignment5.c
//  Description :   Write a program which accepts string from user & reverse the order of the string.
//  Author :        Aryan S. Pardeshi
//  Date :          06/06/2025
//
//////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>

void StrRevX(char *str, char ch)
{
    char *start = NULL;
    char *end = NULL;
    char temp = '\0';

    start = str;
    end = str;

    while(*end != '\0')
    {
        end++;
    }

    end--;

    while(start <= end)
    {
        temp = *start;
        *start = *end;
        *end = temp;

        start++;
        end--;
    }
      
}


int main()
{
    char Arr[50];
    char cValue;


    printf("Enter the String : ");
    scanf("%[^\n]%*c", Arr);

    printf("Enter the Character : ");
    scanf("%c", &cValue);

    StrRevX(Arr, cValue);

    printf("Modified String is :%s",Arr);

    return 0;
}

File: Assignment_53/assignment5.java
/////////////////////////////////////////////////////////////////////////////
//
//     File name   :- assignment5.java
//     Description :- Write a program which accepts two arrays from user and check whether that array and its elements are palindrome or not. 
//     Author name :- Aryan Shailendrasingh Pardeshi
//     Date & Day  :- 13/09/2025 |  SATURDAY
//
/////////////////////////////////////////////////////////////////////////////

import java.util.*;

class MyArray
{
    public boolean isPalindrome(int[] Arr)
    {
        int start = 0;
        int end = Arr.length - 1;

        while (start < end)
        {
            if (Arr[start] != Arr[end])
            {
                return false;
            }
            start++;
            end--;
        }

        return true;
    }

    public int[] CopyArray(int[] Arr1, int[] Arr2)
    {
        int[] newArr = new int[Arr1.length + Arr2.length];

        for (int i = 0; i < Arr1.length; i++)
        {
            newArr[i] = Arr1[i];
        }

        for (int i = 0; i < Arr2.length; i++)
        {
            newArr[Arr1.length + i] = Arr2[i];
        }

        return newArr;
    }
}

class assignment5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter size of first array: ");
        int iSize1 = sobj.nextInt();

        int[] Arr1 = new int[iSize1];
        System.out.println("Enter elements of first array:");

        for (int i = 0; i < iSize1; i++)
        {
            Arr1[i] = sobj.nextInt();
        }

        System.out.print("Enter size of second array: ");

        int iSize2 = sobj.nextInt();

        int[] Arr2 = new int[iSize2];
        System.out.println("Enter elements of second array:");

        for (int i = 0; i < iSize2; i++)
        {
            Arr2[i] = sobj.nextInt();
        }

        MyArray mobj = new MyArray();

        boolean isArr1Palin = mobj.isPalindrome(Arr1);
        boolean isArr2Palin = mobj.isPalindrome(Arr2);

        int[] CombinedArr = mobj.CopyArray(Arr1, Arr2);
        boolean isCombinedPalin = mobj.isPalindrome(CombinedArr);

        System.out.println();

        System.out.println("First array is " + (isArr1Palin ? "a palindrome." : "not a palindrome."));

        System.out.println("Second array is " + (isArr2Palin ? "a palindrome." : "not a palindrome."));

        System.out.println("Combined array is " + (isCombinedPalin ? "a palindrome." : "not a palindrome."));
    }
}
